src/inc/WilErrorReporting.h

```diff
@@ -39,13 +18,29 @@ namespace Microsoft::Console::ErrorReporting
     {
         if (!alreadyReported && FallbackProvider)
         {
-#pragma warning(suppress : 26477 26485 26494 26482 26446) // We don't control TraceLoggingWrite
-            TraceLoggingWrite(FallbackProvider,
-                              "FallbackError",
-                              TraceLoggingKeyword(MICROSOFT_KEYWORD_TELEMETRY),
-                              TelemetryPrivacyDataTag(PDT_ProductAndServiceUsage),
-                              TraceLoggingLevel(WINEVENT_LEVEL_ERROR),
-                              CONSOLE_WIL_TRACELOGGING_FAILURE_PARAMS(failure));
+#pragma warning(suppress : 26477) // Use 'nullptr' rather than 0 or NULL
+            TraceLoggingWrite(
+                FallbackProvider,
+                "FallbackError",
+                TraceLoggingKeyword(MICROSOFT_KEYWORD_TELEMETRY),
+                TelemetryPrivacyDataTag(PDT_ProductAndServiceUsage),
+                TraceLoggingLevel(WINEVENT_LEVEL_ERROR),
+                TelemetryPrivacyDataTag(PDT_ProductAndServicePerformance),
+                TraceLoggingStruct(14, "wilResult", "wilResult"),

```

Comment:
Due to the strict macros, trailing commas in a macro are now forbidden. These can occur with variadic macros like
```cpp
#define FOO(...) foo(123, __VA_ARGS__);
```

If you don't pass any arguments to `FOO` it'll generate a trailing comma, which isn't valid C++ code. One might think trailing commas should be supported in C++, because why not, but they aren't. Instead there's a hackjob called `__VA_OPT__` which lets you write
```cpp
#define FOO(...) foo(123 __VA_OPT__(,) __VA_ARGS__);
```

Since MSVC doesn't support the official `__VA_OPT__`, we have to write `## __VA_ARGS__` from now on, which isn't legit macro syntax, but it works despite using `/Zc:preprocessor`.

---

In this case we don't control `TraceLoggingStruct`, which doesn't use the `## __VA_ARGS__` trick.
As such we have to add a third argument (the "comment") to any TraceLoggingWrite parameters from now on.

Reply:
I suspect that I can accept this. Thanks.

Reply:
We can also set it to the empty string explicitly if we want to save on the .rodata segment. That's what used to happen with the previous non-conformant mode anyways (technically).